tests/test-geometry.lua

```diff
@@ -257,6 +261,134 @@ local steps = {
+        ["NORTH"] =      {10, 0,  100, 80},
+        ["NORTH_EAST"] = {20, 0,  100, 80},
+        ["WEST"] =       {0,  20, 100, 80},
+        ["CENTER"] =     {10, 20, 100, 80},
+        ["EAST"] =       {20, 20, 100, 80},
+        ["SOUTH_WEST"] = {0,  40, 100, 80},
+        ["SOUTH"] =      {10, 40, 100, 80},
+        ["SOUTH_EAST"] = {20, 40, 100, 80},
+        ["STATIC"] =     {10, 30, 100, 80}
+    }
+    for gravity, expectation in pairs(test_data) do
+        gears.table.merge(steps, {
+        function()
+            set_rules {
+                -- All clients will match this rule.
+                { rule = { },properties = {
+                      titlebars_enabled = true,
+                      border_width      = 10,
+                      border_color      = "#00ff00",
+                      size_hints_honor  = false,
+                      x                 = 0,
+                      y                 = 0,
+                      width             = 100,
+                      height            = 100
+                }}
+            }
+            client.connect_signal("request::titlebars", fix_titlebar_size)
+            -- Wait for the previous cleanup to be done
+            if #client.get() == 0 then
+                return true
+            end
+        end,
+        function(count)
+            if count == 1 then
+                print("testing gravity " .. gravity)
+                test_client(nil,nil,nil,nil,nil,{gravity=lgi.Gdk.Gravity[gravity]})
+            else
+                local c = client.get()[1]
+                if c then
+                    assert(c.size_hints.win_gravity == gravity:lower())
+                    assert(c.border_width      == 10  )
+                    assert(c:geometry().x      == 0   )
+                    assert(c:geometry().y      == 0   )
+                    assert(c:geometry().width  == 100 )
+                    assert(c:geometry().height == 100 )
+                    windowid = c.window
+                    c:unmanage()
+                    return true
+                end
+            end
+        end,
+        function()
+            local file = io.popen("xwininfo -id " .. tostring(windowid))
+            local result = file:read("*all")
+            file:close()
+            local x = tonumber(result:match("Absolute upper[-]left X:%s+(%d+)"))
+            local y = tonumber(result:match("Absolute upper[-]left Y:%s+(%d+)"))
+            local w = tonumber(result:match("Width:%s*(%d+)"))
+            local h = tonumber(result:match("Height:%s*(%d+)"))
+            print(gravity, x, y, w, h)
+            assert(x == expectation[1])
+            assert(y == expectation[2])
+            assert(w == expectation[3])
+            assert(h == expectation[4])
+            return true
+        end,
+        function()
+            awful.spawn("wmctrl -c " .. tostring(windowid), false)
+            -- Is there a way to make sure the window is removed?
```

Comment:
What does this do? What do you mean with "make sure the window is removed"?

I *think*  that `wmctrl -c` doesn't do anything. According to https://github.com/dancor/wmctrl/blob/0181e8e4a7ed98255f6ef930875e56560f19aa95/main.c#L738-L740, this sends a `_NET_CLOSE_WINDOW` message to the root window. Such a message is handled by the WM. However, due to your `c:unmanage()` above, AwesomeWM does not know the window any more and I would expect this to do nothing.

Reply:
That's a really good point. I didn't pay much attention to how the close request is handled. I now uses another method to detect window existence - please let me know.